transmogrifier/sources/json/aardvark.py

```diff
@@ -178,58 +179,42 @@ def get_optional_fields(self, source_record: dict) -> dict | None:
-        """Get values from source record for TIMDEX alternate_titles field."""
+    @classmethod
+    def get_alternate_titles(
+        cls, source_record: dict
+    ) -> list[timdex.AlternateTitle] | None:
         return [
             timdex.AlternateTitle(value=title_value)
             for title_value in source_record.get("dct_alternative_sm", [])
-        ]
+        ] or None
 
-    @staticmethod
-    def get_contributors(source_record: dict) -> list[timdex.Contributor]:
-        """Get values from source record for TIMDEX contributors field."""
+    @classmethod
+    def get_content_type(cls, source_record: dict) -> list[str] | None:
+        return source_record.get("gbl_resourceType_sm") or None
+
+    @classmethod
+    def get_contributors(cls, source_record: dict) -> list[timdex.Contributor] | None:
         return [
             timdex.Contributor(value=contributor_value, kind="Creator")
             for contributor_value in source_record.get("dct_creator_sm", [])
-        ]
+        ] or None
 
     @classmethod
-    def get_dates(cls, source_record: dict, source_record_id: str) -> list[timdex.Date]:
-        """Get values from source record for TIMDEX dates field.
-
-        This method aggregates dates from a variety of Aardvark fields.  Once aggregated,
-        the results are filtered to allow only well formed DateRanges or validated date
-        strings.
-        """
-        dates = (
-            cls._issued_dates(source_record)
-            + cls._coverage_dates(source_record)
-            + cls._range_dates(source_record, source_record_id)
-        )
-        return [
-            date
-            for date in dates
-            # skip value validation for DateRange type dates
-            if isinstance(date.range, timdex.DateRange)
-            # validate date string if not None
-            or (date.value is not None and validate_date(date.value, source_record_id))
-        ]
+    def get_dates(cls, source_record: dict) -> list[timdex.Date] | None:
+        dates: list[timdex.Date] = []
+        dates.extend(cls._issued_dates(source_record))
+        dates.extend(cls._coverage_dates(source_record))
+        dates.extend(cls._range_dates(source_record))
+        return dates or None
 
     @classmethod
-    def _issued_dates(cls, source_record: dict) -> list[timdex.Date]:
-        """Get values for issued dates."""
-        issued_dates = []
-        if "dct_issued_s" in source_record:
-            issued_dates.append(
-                timdex.Date(value=source_record["dct_issued_s"], kind="Issued")
-            )
-        return issued_dates
+    def _issued_dates(cls, source_record: dict) -> Iterator[timdex.Date]:
+        if issued_date := source_record.get("dct_issued_s"):  # noqa: SIM102
+            if validate_date(issued_date, cls.get_source_record_id(source_record)):
+                yield (timdex.Date(value=issued_date, kind="Issued"))
```

Comment:
I'm shifting the date validation before  the `Date` instance is created, which seems more inline with how we're doing it in other transforms but again, I want to hear everyone's thoughts on this!

Reply:
I'm good with this pattern!  This is very easy to parse and understand.  With a nod to the [oustanding date parsing issue](https://github.com/MITLibraries/transmogrifier/issues/188), I think this looks great.